client/pages/Dashboard/index.jsx

```diff
@@ -74,17 +74,21 @@ class Dashboard extends Component {
    * @returns {null}
    */
   static getDerivedStateFromProps(nextProps, prevState) {
+    const temporaryState = {};
     if (
       Object.keys(nextProps.activeUser).length > 0
       && nextProps.activeUser !== prevState.activeUser
     ) {
-      return {
-        activeUser: nextProps.activeUser || null,
-        categoryList: nextProps.socialClubs.socialClubs,
-      };
+      temporaryState.activeUser = nextProps.activeUser || null;
     }
     if (nextProps.oauth !== prevState.oauth) {
-      return { oauth: nextProps.oauth };
+      temporaryState.oauth = nextProps.oauth;
+    }
+    if (nextProps.categoryList !== prevState.categoryList) {
+      temporaryState.categoryList = nextProps.categoryList;
+    }
+    if (Object.keys(temporaryState).length) {
```

Comment:
Either add a comment above this condition to state what is happening or change the condition to be a bit readable ?